src/protocol/core.c

```diff
@@ -0,0 +1,52 @@
+#include "zenoh-pico/protocol/core.h"
+
+#include "zenoh-pico/api/primitives.h"
+
+uint8_t _z_id_len(_z_id_t id) {
+    uint8_t len = 16;
+    while (len > 0) {
+        --len;
+        if (id.id[len] != 0) {
+            ++len;
+            break;
+        }
+    }
+    return len;
+}
+_Bool _z_id_check(_z_id_t id) {
+    _Bool ret = false;
+    for (int i = 0; !ret && i < 16; i++) {
+        ret |= id.id[i] != 0;
+    }
+    return ret;
+}
+_z_id_t _z_id_empty() {
+    return (_z_id_t){.id = {
```

Comment:
Please confirm, but I think: `return (_z_id_t){.id = {0}};` would initialize the entire array with 0s.